### PR TITLE
Add comparison operator for Auth::User

### PR DESCRIPTION
**Pull request by chkuang-g**
Add comparison operator for `Auth::User`

***

GHA
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [x] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: auth/src/include/firebase/auth/user.h [hunk 1]
```diff
@@ -198,6 +198,9 @@ class User : public UserInfoInterface {
   /// Assignment operator.
   User& operator=(const User&);
 
+  /// Assignment operator.
```
Comment (jonsimantov) [inline auth/src/include/firebase/auth/user.h:201]: Nit: comment should say "comparison"

**Review by jonsimantov — APPROVED** [review]
Approved with one nit.

File: auth/src/include/firebase/auth/user.h [hunk 2]
```diff
@@ -198,6 +198,10 @@ class User : public UserInfoInterface {
   /// Assignment operator.
   User& operator=(const User&);
 
+  /// Comparison operator.
+  bool operator==(const User&) const;
+  bool operator!=(const User&) const;
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply auth/src/include/firebase/auth/user.h:203]: Please fix n

Comment (chkuang-g) [reply auth/src/include/firebase/auth/user.h:205]: done

[the remark this answers is not in the document]
Comment (chkuang-g) [reply auth/src/include/firebase/auth/user.h:205]: done

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**